docs/06-actors-actions.md

```diff
@@ -0,0 +1,398 @@
+---
+title: Actors
+path: /docs/actors
+---
+
+## Basic actors
+
+For quick and dirty games, you can just create an instance of an `Actor`
+and manipulate it directly.
+
+Actors (and other entities) must be added to a [Scene](/docs/scene) to be drawn
+and updated on-screen.
+
+```ts
+const player = new ex.Actor()
+
+// move the player
+player.vel.x = 5
+
+// add player to the current scene
+game.add(player)
+```
+
+`game.add` is a convenience method for adding an `Actor` to the current scene. The equivalent verbose call is `game.currentScene.add`.
+
+## Actor Lifecycle
+
+An actor has a basic lifecycle that dictates how it is initialized, updated, and drawn. Once an actor is part of a
+[scene](/docs/scene), it will follow this lifecycle.
+
+![Actor Lifecycle](/assets/images/docs/ActorLifecycle.png)
+
+## Extending actors
+
+For "real-world" games, you'll want to `extend` the `Actor` class.
+This gives you much greater control and encapsulates logic for that
+actor.
+
+You can override the [[Actor.onInitialize]] method to perform any startup logic
+for an actor (such as configuring state). `onInitialize` gets called
+**once** before the first frame an actor is drawn/updated. It is passed
+an instance of [Engine](/docs/engine) to access global state or perform coordinate math.
+
+```ts
+class Player extends ex.Actor {
+  public level = 1
+  public endurance = 0
+  public fortitude = 0
+
+  constructor() {
+    super({ x: 50, y: 50 })
+  }
+
+  public onInitialize(engine: ex.Engine) {
+    this.endurance = 20
+    this.fortitude = 16
+  }
+
+  public getMaxHealth() {
+    return 0.4 * this.endurance + 0.9 * this.fortitude + this.level * 1.2
+  }
+}
+```
+
+## Updating actors
```

Comment:
@eonarheim Can you triple sanity check this section to make sure it's correct?

Reply:
No typestate 👎 jk, xstate is good 😄

Reply:
On thing I'd call out is fewer state machines are better, if you can encapsulate the possible states into 1 state it makes debugging much easier. Multiple state machines that influence the game increase the complexity be each machines number of states.